docs/goals_and_soa.md

```diff
@@ -0,0 +1,392 @@
+# Design principles
+
+## Why Cardano DApp modeling language is important?
+
+While protototypes can be easily constructed with existing Cardano frameworks, making it secure and production ready is sisyphean task.
+
+We believe, that onchain PL improvements by themself could not change that. Only higher-level modeling framework can lead to reliable DApp production.
+
+In following enumerate our high-level design goals
+and demonstrate them by list of specific problems
+arising in existing DApps codebase.
+Such problems are specific security vulnerabilies types
+and various kinds development, audit and support cost increase.
+After that we examine how existing solutions
+are covering our high-level goals.
+
+## High-level goals
+
+1. DApp logic as whole (synced-by-construction)
+2. Code is free from common security weaknesses by construction (secure-by-construction)
+3. Seamplessly emulate and test anything (emulate-anything)
+4. Declarativity close to informal specification and bridging lightweight formal methods (declarative-spec)
+5. Generally production ready (production-ready)
+
+## Not in scope at all
+
+* Composable inter-DApp interactions and DApp migrations
+* Scheduled actions
+
+## Not in scope for current Catalyst funding
+
+* CTL and overall frontend transaction construction model
```

Comment:
How does it differ from mentioned earlier "offchain Tx construction and indexing backend"? Do we cover any other means to generate off-chain code?

Reply:
> indexing backend

It actually get out of scope for Milestones accepted.

> Do we cover any other means to generate off-chain code?

There would be no off-chain code generation. Only typeclasses.